File: networking/async1.cpp
#include <iostream>
#include <thread>
#include <chrono>

void task(const char* name, int sec) {
    std::cout << name << " started\n";
    std::this_thread::sleep_for(std::chrono::seconds(sec));
    std::cout << name << " finished after " 
                      << seconds << " seconds\n";
}

int main() {
    std::thread t1(task, "Task A", 3);
    std::thread t2(task, "Task B", 1);
    std::thread t3(task, "Task C", 4);
    std::thread t4(task, "Task D", 2);

    t1.join();
    t2.join();
    t3.join();
    t4.join();

    return 0;
}

File: networking/async1.py
import asyncio

async def task(name, delay):
    print(f"{name} started")
    await asyncio.sleep(delay) 
    print(f"{name} finished after {delay} seconds")

async def main():
    task1 = asyncio.create_task(task("Task A", 3))
    task2 = asyncio.create_task(task("Task B", 1))
    task3 = asyncio.create_task(task("Task C", 4))
    task4 = asyncio.create_task(task("Task D", 2))

    await task1
    await task2
    await task3
    await task4

asyncio.run(main())
